vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,352 @@
+package vcd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdOrgUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdOrgUserCreate,
+		Read:   resourceVcdOrgUserRead,
+		Delete: resourceVcdOrgUserDelete,
+		Update: resourceVcdOrgUserUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdOrgUserImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCase("lower"),
+				Description:  "User's name. Only lowercase letters allowed. Cannot be changed after creation",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization this user belongs to",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "Role within the organization",
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+
+				ConflictsWith: []string{"password_file"},
+				Description: "The user's password. This value is never returned on read. " +
```

Comment:
Please elaborate, if value is not returned on read, is it reporting that there's a change with each succeeding `plan`?

Reply:
The value is not reported by vCD, but the plan is based on the values in the configuration file.
If you change the value of the password from an existing string to empty text, the plan reports a change that will not correspond to a real change in vCD. 
The way of using a password (as opposed to a password file) is to set the password, then re-run with an empty password.
From that point on, until you change the password using Terraform, the plan will mean what it should.

Summing up: when changing the password to a different text, there will be a real change and a plan change. When changing a text to empty string (or removing the password field) there will be a change in the plan, but it will be a dud in vCD (meaning an update without changes and without errors).

Reply:
If you know how to make such fieldDiffFunc, let me know. The ones I tried would suppress the password before it reaches the provider.

Reply:
DiffSuppressFunc: func(k string, old string, new string, d *schema.ResourceData) bool {
return old == ""
}

Reply:
This one will prevent the password from reaching the provider, and fails on creation